tests/unit/challenges/test_aws_utils.py

```diff
@@ -0,0 +1,199 @@
+            username="someparticipantuser",
+            email="participantuser@test.com",
+            password="secret_password",
+        )
+
+        EmailAddress.objects.create(
+            user=self.participant_user,
+            email="participantuser@test.com",
+            primary=True,
+            verified=True,
+        )
+
+        self.challenge_host_team = ChallengeHostTeam.objects.create(
+            team_name="Test Challenge Host Team", created_by=self.user
+        )
+
+        self.challenge = Challenge.objects.create(
+            title="Test Challenge",
+            short_description="Short description for test challenge",
+            description="Description for test challenge",
+            terms_and_conditions="Terms and conditions for test challenge",
+            submission_guidelines="Submission guidelines for test challenge",
+            creator=self.challenge_host_team,
+            published=False,
+            is_registration_open=True,
+            enable_forum=True,
+            queue="test_queue",
+            anonymous_leaderboard=False,
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+            approved_by_admin=False,
+        )
+        self.challenge.slug = "{}-{}".format(
+            self.challenge.title.replace(" ", "-").lower(), self.challenge.pk
+        )[:199]
+        self.challenge.save()
+
+        self.challenge_host = ChallengeHost.objects.create(
+            user=self.user,
+            team_name=self.challenge_host_team,
+            status=ChallengeHost.ACCEPTED,
+            permissions=ChallengeHost.ADMIN,
+        )
+        self.challenge2 = Challenge.objects.create(
+            title="Test Challenge 2",
+            creator=self.challenge_host_team,
+            queue="test_queue_2",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+
+        self.client.force_authenticate(user=self.user)
+        self.ecs_client = boto3.client(
+            "ecs",
+            region_name=os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+            aws_secret_access_key=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+            aws_access_key_id=os.environ.get("AWS_ACCESS_KEY_ID"),
+        )
+
+    def set_challenge_workers(self, challenge, num_workers):
+        challenge.workers = num_workers
+        challenge.save()
+
+
+@mock_ecs
+class BaseAdminCallsClass(BaseTestClass):
+    def setUp(self):
+        super(BaseAdminCallsClass, self).setUp()
+
+        self.challenge3 = Challenge.objects.create(
+            title="Test Challenge 3",
+            creator=self.challenge_host_team,
+            queue="test_queue_3",
+            start_date=timezone.now() - timedelta(days=2),
+            end_date=timezone.now() + timedelta(days=1),
+        )
+        self.ecs_client.create_cluster(clusterName="cluster")
+        self.client_token = "abc123"
```

Comment:
@nikochiko can we use a randomly generated string here

Reply:
Done :+1: